analytics-kotlin/src/main/java/com/segment/analytics/utilities/JSON.kt

```diff
@@ -78,11 +78,21 @@ fun JsonPrimitive.toContent(): Any? {
 }
 
 // Utility function to retrieve a boolean value from a jsonObject
-fun JsonObject.getBoolean(key: String): Boolean? = this[key]?.jsonPrimitive?.boolean
+fun JsonObject.getBoolean(key: String): Boolean? = this[key]?.jsonPrimitive?.booleanOrNull
 
 // Utility function to retrieve a string value from a jsonObject
 fun JsonObject.getString(key: String): String? = this[key]?.jsonPrimitive?.contentOrNull
 
+// Utility function to retrieve a double value from a jsonObject
+fun JsonObject.getDouble(key: String): Double? = this[key]?.jsonPrimitive?.doubleOrNull
+
+// Utility function to retrieve a int value from a jsonObject
+fun JsonObject.getInt(key: String): Int? = this[key]?.jsonPrimitive?.intOrNull
+
 // Utility function to retrieve a string set (from jsonArray) from a jsonObject
 fun JsonObject.getStringSet(key: String): Set<String>? =
     this[key]?.jsonArray?.map { it.jsonPrimitive.content }?.toSet()
+
+fun JsonObject.getMapSet(key: String): Set<Map<String, JsonElement>>? {
```

Comment:
any reason why we are returning a `Set<Map<String, JsonElement>>` instead of `Set<Map<String, Any>>`?

Reply:
could potentially do this 
```
fun JsonObject.getMapSet(key: String): Set<Map<String, Any?>>? =
    this[key]?.jsonObject?.map { mapOf(it.key to it.value.jsonPrimitive.toContent()) }?.toSet()
```

Reply:
I'm game for that